src/commit.c

```diff
@@ -302,8 +298,8 @@ int apk_solver_commit_changeset(struct apk_database *db,
 					   "The following packages will be upgraded");
 			r += dump_packages(changeset, cmp_reinstall,
 					   "The following packages will be reinstalled");
-			printf("After this operation, %zd %ciB of %s.\n",
-				(size_diff < 0) ? -size_diff : size_diff,
+			printf("After this operation, %lld %ciB of %s.\n",
```

Comment:
This fails now if the size is less than 1024. As get_human_unit() returns 'B' as unit, and you'd print BiB which is no sense. I think apk_get_human_unit should return string "B", "KiB", ... Or be fixed to return at least KiB...

Reply:
> As get_human_unit() returns 'B' as unit, and you'd print BiB which is no sense.

Oh, indeed. Good catch!

> I think apk_get_human_unit should return string "B", "KiB", ... Or be fixed to return at least KiB...

What do you prefer? Returning a string would either require doing memory allocations in `apk_get_human_unit` or passing a destination buffer (allocated on the caller stack) to it.

Returning KiB as the smallest unit would require less changes and I don't think we have any packages smaller than 1 KiB. That being said, I am personally fine with either solution.

Reply:
Returning string literal `"KiB"` as `const char *` is perfectly ok. Though, then the function should indicate that it's for size units. I think I would prefer this. Mostly because then the unit type comes from the function instead of the format string. In case we would later want to support both KB and KiB which is switchable via a flag or similar.

Reply:
> Returning string literal `"KiB"` as `const char *` is perfectly ok.

I agree, implemented this accordingly.

> Though, then the function should indicate that it's for size units.

What do you mean by that?

Reply:
Function name should indicate it. E.g. `apk_get_human_size`. Bit that's minor. And the commits need to be flattened to get rid of the fixups. But otherwise looks good now.  Thanks!

Reply:
Done.